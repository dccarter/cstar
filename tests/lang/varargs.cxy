func varargs(...args: auto) {
}

func add(...nums: f32) {
    var sum : f32 = 0;
    #const i = 0;
    #for (const num: nums) {
        sum += nums.#{i}
        #{i += 1}
    }
}

func many(x: i32, ...rest: f32) {
    add(<f32>x, ...rest)
}

func main(args: [string])
{
    varargs(10, "Hello", 'c', 2000.6)
    many(10, 10.0, 100.6)
}