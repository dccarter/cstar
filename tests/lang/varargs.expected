func varargs(...args: ) {}

func add(...nums: f32) {
    const sum : f32 = 0
    var i = 0
    for (var num : nums.Tinfo) {
        sum += nums.i
        i += 1
    }
}

func many(x: i32, ...rest: f32) {
    add(<f32>x, ...rest)
}

func varargs_0(...args: (i32, string, wchar, f32)) {}

func many_1(x: i32, ...rest: (f32, f32)) {
    add_2(<f32>x, rest.0, rest.1)
}

func add_2(...nums: (f32, f32, f32)) {
    const sum : f32 = 0
    
    {
        sum += nums.0
        
    }
    {
        sum += nums.1
        
    }
    {
        sum += nums.2
        
    }
}

func main(args: [string]) {
    varargs_0(10, "Hello", 'c', 2.000600e+03)
    many_1(10, (1.000000e+01, 1.006000e+02))
}
